R/build-logo.R

```diff
@@ -7,19 +7,36 @@ build_logo <- function(pkg = ".") {
 
   file_copy_to(pkg, logo_path, from_dir = path_dir(logo_path))
 
-  if (!requireNamespace("magick", quietly = TRUE)) {
-    message("magick not available, using default pkgdown favicon.ico")
+  cat_line("Creating ", dst_path("favicon.ico"))
 
-    if (!file_exists(path(pkg$dst_path, "favicon.ico"))) {
-      file_copy(path_pkgdown("assets", "favicon.ico"), pkg$dst_path)
-    }
-    return()
-  }
+  logo_string <- readBin(logo_path, what = "raw", n = fs::file_info(logo_path)$size) %>%
```

Comment:
I'd also avoid the pipe here, since it doesn't improve readability much, and makes tracebacks a bit harder to read.